src/main/kotlin/com/hyperdevs/poeditor/gradle/network/api/PoEditorApiModels.kt

```diff
@@ -25,29 +25,91 @@ import java.util.Date
  */
 open class PoEditorResponse(open val response: ResponseStatus)
 
+/**
+ * Basic response data.
+ */
+data class ResponseStatus(val status: String,
+                          val code: String,
+                          val message: String)
+
 /**
  * PoEditor response to "list languages" call.
  */
-data class ListProjectLanguagesResponse(override val response: ResponseStatus,
-                                        val list: List<ProjectLanguage>) : PoEditorResponse(response)
+data class ListLanguagesResponse(override val response: ResponseStatus,
+                                 val result: ListLanguagesResult) : PoEditorResponse(response)
+
+/**
+ * Result of a "list language" call.
+ */
+data class ListLanguagesResult(val languages: List<ProjectLanguage>)
 
 /**
  * PoEditor response to "export languages" call.
  */
 data class ExportResponse(override val response: ResponseStatus,
-                          val item: String) : PoEditorResponse(response)
+                          val result: ExportResult) : PoEditorResponse(response)
 
 /**
- * Basic response data.
+ * Result of a "list language" call.
  */
-data class ResponseStatus(val status: String,
-                          val code: String,
-                          val message: String)
+data class ExportResult(val url: String)
 
 /**
  * Information about a language in PoEditor.
  */
 data class ProjectLanguage(val name: String,
                            val code: String,
+                           val translations: Int,
                            val percentage: Double,
-                           val updated: Date?)
\ No newline at end of file
+                           val updated: Date?)
+
+/**
+ * Types of file export allowed in PoEditor.
+ */
+enum class ExportType {
+    PO,
+    POT,
+    MO,
+    XLS,
+    XLSX,
+    CSV,
+    INI,
+    RESW,
+    RESX,
+    ANDROID_STRINGS,
+    APPLE_STRINGS,
+    XLIFF,
+    PROPERTIES,
+    KEY_VALUE_JSON,
+    JSON,
+    YML,
+    XLF,
+    XMB,
+    XTB,
+    ARB,
+    RISE_360_XLIFF;
+
+    companion object {
+        /** Returns the enum value associated to a string value. */
+        fun from(filterString: String) = valueOf(filterString.toUpperCase())
```

Comment:
Actually - not sure if I understand it right, but is this from() method used for deserialization? In that case as an alternative you could annotate all fields with their string representation. 
Could also make the field that stores the enum nullable (to handle the case of new strings that have no representation in the enum). 
@Json(name = "thing")

It's just an option, it's up to you, of course. Maybe you decided against it to avoid having all the annotations.

Reply:
Yeah, it's intended :P. It may be a bit overkill to add a typed exception just for the use case, I think the `IllegalArgumentException` should give enough feedback about what's happening (which should be an incorrect value from parameters, I hope)

Regardless, thanks for pointing it out!